src/libstd/dynamic_lib.rs

```diff
@@ -53,8 +52,8 @@ impl DynamicLibrary {
 
     /// Lazily open a dynamic library. When passed None it gives a
     /// handle to the calling process
-    pub fn open(filename: Option<&Path>) -> Result<DynamicLibrary, String> {
-        let maybe_library = dl::open(filename.map(|path| path.as_vec()));
+    pub fn open(filename: Option<&path::Path>) -> Result<DynamicLibrary, String> {
+        let maybe_library = dl::open(filename.map(|path| path.as_path().as_os_str()));
```

Comment:
This can probably just be `path.as_os_str()` instead of going through `as_path()`